### ISSUE TITLE
Material "Durations" class?

### ISSUE THREAD

**Issue by brianegan** [flutter/flutter#5735]
Not a bug, just a suggestion! Happy to help implement it as well, if it's something you would be interested in :)

I really enjoy the `Icons`, `Colors`, etc classes available in the `material` package. They helped me bust out a quick, decent looking prototype in no time at all!

I also started using the `AnimatedOpacity` class to perform a couple of transitions, which required a `Duration` parameter, and wondered: "Oh yes, what exactly are the correct Material durations?" 

Luckily, the material specs are well documented here: https://material.google.com/motion/duration-easing.html with the specs. I created a small helper class in my app, but wondered if they could be codified as a class in the `Material` package?

Some constants already exist in the Material package (https://docs.flutter.io/flutter/material/material-library.html), but the material guidelines require extending the animation duration by 30% for Tablets, for example.

Would it make sense to have another class, such as `Durations`, with methods that take something like a `Size` (preferably something better), and return the correct duration?

This could look something like:

```dart
class Durations {
  static final kStandardMobileDurationInMs = 300;
  static final kEnterScreenMobileDurationInMs = 225;
  static final kExitScreenMobileDurationInMs = 195;

  static Duration standardDuration({ Size screenSize }) {
    return new Duration(
      milliseconds:
      (kStandardMobileDurationInMs * screenSizeModifier(screenSize)).round()
    );
  }

  static double screenSizeModifier(Size screenSize) {
    if (screenSize == null) {
      return 1.0;
    } else if (isLargeScreen(screenSize)) {
      return 1.3;
    } else if (isSmallScreen(screenSize)) {
      return 0.7;
    }

    return 1.0;
  }

  static bool isSmallScreen(Size screenSize) {
    return screenSize.width < 320 || screenSize.height < 320;
  }

  static bool isLargeScreen(Size screenSize) {
    return screenSize.width > 500 || screenSize.height > 500;
  }

  // Pseudo-code / not implemented, but extra durations from the spec as an example
  static Duration complexFullScreenDuration();
  static Duration enterScreenDuration();
  static Duration exitScreenDuration();
}
```

Let me know whatcha think!

**Comment by eseidelGoogle** [comment]
@Hixie or @abarth or @HansMuller would know.  Sounds useful.

**Comment by abarth** [comment]
Yes, something like that sounds like a good idea.  I'm not sure about the details exactly, but it looks like a core building block for building up proper adaptive animations.

**Comment by abarth** [comment]
Removing from API stability because this is about adding APIs, not about breaking existing APIs.

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/issues/5553 in case it's related.

**Comment by Piinks** [comment]
We are closing this suggestion as obsolete. While a centralized `Durations` class was a useful proposal in the early days of the framework, modern Flutter (especially with Material 3) has integrated durations more deeply into the theming system. Durations are now context-specific and can be accessed through appropriate theme data or are defined as constants within specific widget implementations, providing a more robust and adaptive approach than a single global helper class.

If there is more work to do here, please let us know by filing a new issue with up to date information. Thanks!
